mcs/build/library.make

```diff
@@ -295,24 +295,37 @@ endif
 # The library
 
 # If the directory contains the per profile include file, generate list file.
-PROFILE_sources := $(firstword $(if $(PROFILE_PLATFORM),$(wildcard $(PROFILE_PLATFORM)_$(PROFILE)_$(LIBRARY).sources)) $(wildcard $(PROFILE)_$(LIBRARY).sources) $(wildcard $(LIBRARY).sources))
-PROFILE_excludes = $(firstword $(if $(PROFILE_PLATFORM),$(wildcard $(PROFILE_PLATFORM)_$(PROFILE)_$(LIBRARY).exclude.sources)) $(wildcard $(PROFILE)_$(LIBRARY).exclude.sources))
+SOURCES_PLATFORM ?= $(PROFILE_PLATFORM)
+PROFILE_sources := $(firstword $(if $(SOURCES_PLATFORM),$(wildcard $(SOURCES_PLATFORM)_$(PROFILE)_$(LIBRARY).sources)) $(wildcard $(PROFILE)_$(LIBRARY).sources) $(wildcard $(LIBRARY).sources))
+PROFILE_excludes = $(firstword $(if $(SOURCES_PLATFORM),$(wildcard $(SOURCES_PLATFORM)_$(PROFILE)_$(LIBRARY).exclude.sources)) $(wildcard $(PROFILE)_$(LIBRARY).exclude.sources))
+
+monolite_path := $(topdir)/class/lib/monolite-$(BUILD_PLATFORM)/$(MONO_CORLIB_VERSION)
+
+with_mono_path_monolite = MONO_PATH="$(monolite_path)$(PLATFORM_PATH_SEPARATOR)$(monolite_path)/Facades$(PLATFORM_PATH_SEPARATOR)$$MONO_PATH"
+
+gensources = $(topdir)/build/gensources.exe
+$(gensources): $(topdir)/build/gensources.cs
+	echo "Compiling gensources.exe"
+	$(with_mono_path_monolite) $(RUNTIME) $(monolite_path)/mcs.exe -debug -out:$(gensources) $(topdir)/build/gensources.cs
 
 # Note, gensources.sh can create a $(sourcefile).makefrag if it sees any '#include's
 # We don't include it in the dependencies since it isn't always created
-sourcefile = $(depsdir)/$(PROFILE_PLATFORM)_$(PROFILE)_$(LIBRARY_SUBDIR)_$(LIBRARY).sources
-$(sourcefile): $(PROFILE_sources) $(PROFILE_excludes) $(topdir)/build/gensources.sh $(depsdir)/.stamp
-	$(SHELL) $(topdir)/build/gensources.sh $@ '$(PROFILE_sources)' '$(PROFILE_excludes)'
+sourcefile = $(depsdir)/$(SOURCES_PLATFORM)_$(PROFILE)_$(LIBRARY_SUBDIR)_$(LIBRARY).sources
+$(sourcefile): $(PROFILE_sources) $(PROFILE_excludes) $(topdir)/build/gensources.sh $(depsdir)/.stamp $(gensources)
+	$(SHELL) $(topdir)/build/gensources.sh "$@.shout" '$(PROFILE_sources)' '$(PROFILE_excludes)'
```

Comment:
Why are we still using both `gensources.sh` and `gensources.exe`? I understand it's useful during development to guarantee it generates the same thing, but once you verified it, you should not commit both. Otherwise, you should add a `diff "/tmp/$(LIBRARY).csgensources" "/tmp/$(LIBRARY).shgensources"` and guarantee it's the same.